### ISSUE TITLE
Save page without images / replace image by placeholder

### ISSUE THREAD

**Issue by gety9** [gildas-lormeau/SingleFile#839]
**Is your feature request related to a problem? Please describe.**
I am using SingleFile's extension for FF and would like to be able to save page WITHOUT images
(cause most of the time they don't provide any informational value and just take lots of space)

**Describe the solution you'd like**
Save page as single file but instead of image have nothing or gray place holder

**Describe alternatives you've considered (optional)**
I tried select all text on page excluding images and then using "Save selection" but it's tricky cause often image goes right between text blocks. I also tried turning page into read mode, but then SingleFile don't work.

---------------------

2 more questions which don't warrant opening separate issue.
1. In file name template is it possible to have title without spaces? I mean to make {page-title} produce ThisIsAritcleTitle instead of This is article title?
2. Where can i find full manual? I found lots of useful features like https://github.com/gildas-lormeau/SingleFile/issues/736#issuecomment-898973316 https://github.com/gildas-lormeau/SingleFile/issues/814#issuecomment-969448180 by searching issues, but they are no where in Read.me or FAQ.

**Comment by abolibibelot1980** [comment]
I concur that this would be a welcome feature. (I've created an [issue report about a year ago regarding the storing of identical images within the same page](https://github.com/gildas-lormeau/SingleFile/issues/578), I don't know if it has improved on that matter since then.)

Regarding the name template without space, I don't think so (I would guess that it would be fairly more complex to implement and it would be of "niche" interest), but it's easy to rename saved pages with any batch renaming utility (I generally use Lupas Rename which is old but streamlined and reliable, and Bulk Rename Utility for more complex cases).

**Comment by gildas-lormeau** [comment]
This is typically why I added a way to customize the way SingleFile works via a userscript, see https://github.com/gildas-lormeau/SingleFile/wiki/How-to-execute-a-user-script-before-a-page-is-saved. For example, this script would do the job (it only removes "foreground" images though):
```js
Array.from(document.images).forEach(image => { image.src = image.srcset = ""; })
```

I confirm that you cannot remove spaces easily. However I guess you could add it in the list of replaced characters (`filenameReplacedCharacters`), and set "" as replacement character (`filenameReplacementCharacter`). These are hidden options, see https://github.com/gildas-lormeau/SingleFile/wiki/Hidden-options for more information.

There is no other manual than the help page in the extension. As far as I know, the 2 features you're referring to are documented in the help page though.

**Comment by gildas-lormeau** [comment]
Regarding the removal of spaces, an alternative could also be to run a userscript (see last example here: https://github.com/gildas-lormeau/SingleFile/wiki/How-to-execute-a-user-script-before-a-page-is-saved)

**Comment by miku86** [comment]
This would be an awesome feature.

I like to archive travel blogs, especially how to deal with stuff and interesting insights.
Many bloggers add a lot of images to these articles.

E.g. the page I just downloaded has a size of 93MB, the images are 91MB.
The file is actually so big that sed, vim and vscode crash.

I was able to handle it with your script, but I know a lot of non-dev user that use your extension.

Let us know if/how we can proceed with this.

Thanks
Michael
